server/netty-server/src/main/scala/sttp/tapir/server/netty/internal/NettyServerHandler.scala

```diff
@@ -40,6 +47,40 @@ class NettyServerHandler[F[_]](route: Route[F], unsafeRunAsync: (() => F[Unit])
     res
   }
 
+  override def handlerAdded(ctx: ChannelHandlerContext): Unit =
+    if (ctx.channel.isActive) {
+      initHandler(ctx)
+    }
+  override def channelActive(ctx: ChannelHandlerContext): Unit = initHandler(ctx)
```

Comment:
is it so that either `channelActive` OR `handlerAdded` is called? won't both be called?

Reply:
Indeed sometimes both can be called, but now I've added a check in initHandler which will prevent double listener registration.